### DISCUSSION TITLE
MongoDB id/_id mismatch

### DISCUSSION THREAD

**Discussion by luisrudge** [dotansimha/graphql-code-generator#5857] [Q&A] [answered]
Hi, friends. I'm trying the mongodb plugin but I think I'm doing something wrong hehe.
As you can see, the default generation will never be valid, because the id/_id mismatch. Also, the _id is an ObjectId, which makes the 'where' clause a bit of a mess as well.

![](https://storage.crisp.chat/users/upload/visitor/14c17513d2982f0/image_1cbh0mk.png)

If I change the `objectIdType` to be a string, then I have to change all queries to create ObjectIds in the where clause.

Am I missing something?

**Comment by dotansimha** [comment, marked as answer]
You need a resolver for that... The types are only utilities, you need a resolver to map the object field correctly :)
